app/models/concerns/links_generator.rb

```diff
@@ -0,0 +1,24 @@
+module LinksGenerator
+  extend ActiveSupport::Concern
```

Comment:
I'd rather use Ruby `self.included`, `self.extended` hooks as opposed to Concerns but that's debatable so feel free to read on the topic and decide for yourself 🙂